modules/rtprint.php

```diff
@@ -146,11 +146,31 @@
         if ($subject) {
             $where[] = 't.subject ?LIKE? '.$DB->Escape("%$subject%");
         }
-        $catids = (is_array($categories) ? array_keys($categories) : null);
-        if (!empty($catids)) {
-            $where[] = 'tc.categoryid IN ('.implode(',', $catids).')';
-        } else {
-            $where[] = 'tc.categoryid IS NULL';
+
+        // category id's
+        if (!empty($categories)) {
+            if (!is_array($categories)) {
+                $categories = array($categories);
+            }
+
+            if (in_array('all', $categories)) {
+                $filter['catids'] = null;
+            } else {
+                $filter['catids'] = Utils::filterIntegers($categories);
+                if (in_array(-1, $filter['catids'])) {
+                    if (count($filter['catids']) > 1) {
+                        $catidsfilter = '(';
+                    }
+                    $catidsfilter .= 'tc.categoryid IS NULL';
+                    $filter['catids'] = array_diff($filter['catids'], ["-1"]);
+                    if (!empty($filter['catids'])) {
+                        $catidsfilter .= ' OR tc.categoryid IN (' . implode(',', $filter['catids']) . '))';
+                    }
+                    $where[] = $catidsfilter;
+                } else {
+                    $where[] = 'tc.categoryid IN (' . implode(',', $filter['catids']) . ')';
+                }
+            }
```

Comment:
Wymuszenie sprawdzenia zgłoszeń bez kategorii już staje się niepotrzebne:
```
$where[] = 'tc.categoryid IS NULL';
```
?
Wydaje mi się, że proponowana zmiana niejako przy okazji pozbawi nas tego feature-a.

No i zasadnicza sprawa - przepisywanie wszystkiego kompletnie od zera przy każdej drobnej zmianie do niczego sensownego nie prowadzi poza zwiększeniem wszystkim nakładu pracy. Prawie pewne, że to spowoduje jakieś błędy w działaniu, a potem będziemy znowu przez ileś wersji to prostowali. O wiele mniej błędów pojawia się, gdy
zmieniasz tylko mniej znaczący kawałek kodu.

Reply:
1. Zgadzam się z tym że nie powinno przepisywać się kodu od zera ale zwróć uwagę, że obecny raport nie dawał możliwości wybrania zgłoszeń "bez kategorii" lub "bez kategorii i z jakąś kategorią".

2. Co do tej linijki - algorytm jest taki:
- sprawdź czy array zawiera słowo all (brak warunku kategorii),
- sprawdź czy array zawiera -1 (bez kategorii),
- wrzuć nr kategorii w tą linijkę którą chcesz zamienić,
Także nie da rady tego tak zmienić.

Reply:
>  Co do tej linijki - algorytm jest taki:

Czy to znaczy, że nowa wersja znajdzie również zgłoszenia do których nie jest przypisana żadna kategoria?
Jaką wartość będzie miało wówczas array? -1 jako element tablicy?

Reply:
Dokładnie tak.

Utworzony fragment zapytania będzie miał postać:
```
WHERE 1 = 1  **AND tc.categoryid IS NULL** ORDER BY t.createtime
```

Dodatkowo możesz wybór mieć "nowa kategoria lub inna kategoria".

Wtedy ten fragment będzie miał postać
```
WHERE 1 = 1  **AND (tc.categoryid IS NULL OR tc.categoryid IN (4))** ORDER BY t.createtime
```